### ISSUE TITLE
[iOS] postprocess build overwrites LIBRARY_SEARCH_PATH

### ISSUE THREAD

**Issue by NumaNumaNuma** [OneSignal/OneSignal-Unity-SDK#233]
**Description:**

Not entirely sure if this one is due to OneSignal or Unity but after building, the XCode project can't find its libraries because the build settings LIBRARY_SEARCH_PATHS has been overwritten. If the post process build is commented out, everything is fine. This is new and didn't happen in 2019.2.0f1
(I haven't tested any other versions in between, so it might happen in more recent ones too)

**Environment**
OneSignal SDK 2.9.0
Unity 2019.2.8f1
XCode 11.0

**Steps to Reproduce Issue:**

1. Build an empty project in Unity 2019.8.0f1
2. build settings LIBRARY_SEARCH_PATHS look normal:
```
$(inherited)
"$(SRCROOT)"
"$(SRCROOT)/Libraries"
```

3. Add OneSignal plugin to the project and rebuild
4. LIBRARY_SEARCH_PATHS now only contains the OneSignal path:
```
$(inherited)
$(PROJECT_DIR)/Libraries/OneSignal/Platforms/iOS
```
5. Comment out this line and everything is fine (but onesignal library is missing from onesignal target of course):
`contents = contents.Replace ("PRODUCT_BUNDLE_IDENTIFIER = ", "LIBRARY_SEARCH_PATHS = (\n\t\t\t\t\t\"$(inherited)\",\n\t\t\t\t\t\"$(PROJECT_DIR)/Libraries/OneSignal/Platforms/iOS\",\n\t\t\t\t);\nPRODUCT_BUNDLE_IDENTIFIER = ");`

6. Try the same thing in 2019.2.0f1, no issues.

Thanks!

**Comment by NumaNumaNuma** [comment]
Workaround: remove the offending line and use this instead (placed higher up in the code):

project.AddBuildProperty(notificationServiceTarget, "LIBRARY_SEARCH_PATHS", "$(PROJECT_DIR)/Libraries/OneSignal/Platforms/iOS");

This worked fine for me.

**Comment by rgomezp** [comment]
Howdy,
Thanks for bringing this up and providing a workaround. We will look into this when we get a chance. Thanks for your patience

**Comment by ArnoBen** [comment]
@NumaNumaNuma Where should I write this workaround ? Is it in Unity in the file PostProcessBuildPlayer_iOS.cs ?
@rgomezp Has this issue been fixed since the last post ? I came here from the error "library not found for -liPhone-lib" and even though I installed the last version of OneSignal, I still get the error.
Unity 2018.4.11f1

Thanks in advance.

**Comment by NumaNumaNuma** [comment]
@ArnoBen 
Put it at the end of the OnPostProcessBuild method, just before writing the project file:

```
// Add OneSignal lib
project.AddBuildProperty(notificationServiceTarget, "LIBRARY_SEARCH_PATHS", "$(PROJECT_DIR)/Libraries/OneSignal/Platforms/iOS");
```

**Comment by ArnoBen** [comment]
@NumaNumaNuma Thanks for your answer. I did exactly that and still have "Library not found for -liPhone-lib" error when building from Xcode.
According to [this post](https://forum.unity.com/threads/got-library-not-found-for-liphone-lib-error-after-unity-update.759932/), it's a problem on Unity's side.

**Comment by ceceomer** [comment]
> @NumaNumaNuma Thanks for your answer. I did exactly that and still have "Library not found for -liPhone-lib" error when building from Xcode.
> According to [this post](https://forum.unity.com/threads/got-library-not-found-for-liphone-lib-error-after-unity-update.759932/), it's a problem on Unity's side.

Same here, this post isn't a solution for me. I am still getting same error.

**Comment by NumaNumaNuma** [comment]
Depending on the order in which each of your plugins' post build scripts are executed you might be missing some other folders too. Just make sure (manually or via a post build script) that the correct folders are listed in the Library Search Paths in XCode: (Build Settings -> Search Paths -> Library Search Paths). The build error should tell you which library isn't found, find in which folder it is and add the path to xcode.  

Most likely you just need to add:
"$(SRCROOT)"
"$(SRCROOT)/Libraries"
$(SRCROOT)/Libraries/Plugins/iOS

**Comment by ceceomer** [comment]
> "$(SRCROOT)"
> "$(SRCROOT)/Libraries"
> $(SRCROOT)/Libraries/Plugins/iOS

I added them manually to xcode but still same error...

**Comment by bilck** [comment]
Just fixed this issue by changing line 194 on "Assets/OneSignal/Editor/PostProcessBuildPlayer_iOS.cs" file:
Changed from:
`contents = contents.Replace ("PRODUCT_BUNDLE_IDENTIFIER = ", "LIBRARY_SEARCH_PATHS = (\n\t\t\t\t\t\"$(inherited)\",\n\t\t\t\t\t\"$(PROJECT_DIR)/Libraries/OneSignal/Platforms/iOS\",\n\t\t\t\t);\nPRODUCT_BUNDLE_IDENTIFIER = ");`
To:
`contents = contents.Replace ("PRODUCT_BUNDLE_IDENTIFIER = ", "LIBRARY_SEARCH_PATHS = (\n\t\t\t\t\t\"$(inherited)\",\n\t\t\t\t\t\"$(PROJECT_DIR)/Libraries/OneSignal/Platforms/iOS\",\n\t\t\t\t\t\"$(SRCROOT)/Libraries/**\",\n\t\t\t\t);\nPRODUCT_BUNDLE_IDENTIFIER = ");`

As a suggestion, why don't you guys change "InsertStaticFrameworkIntoTargetBuildPhaseFrameworks" method to properly use PBXProject class and properly change build properties?

**Comment by IsmaelNascimento** [comment]
> Just fixed this issue by changing line 194 on "Assets/OneSignal/Editor/PostProcessBuildPlayer_iOS.cs" file:
> Changed from:
> `contents = contents.Replace ("PRODUCT_BUNDLE_IDENTIFIER = ", "LIBRARY_SEARCH_PATHS = (\n\t\t\t\t\t\"$(inherited)\",\n\t\t\t\t\t\"$(PROJECT_DIR)/Libraries/OneSignal/Platforms/iOS\",\n\t\t\t\t);\nPRODUCT_BUNDLE_IDENTIFIER = ");`
> To:
> `contents = contents.Replace ("PRODUCT_BUNDLE_IDENTIFIER = ", "LIBRARY_SEARCH_PATHS = (\n\t\t\t\t\t\"$(inherited)\",\n\t\t\t\t\t\"$(PROJECT_DIR)/Libraries/OneSignal/Platforms/iOS\",\n\t\t\t\t\t\"$(SRCROOT)/Libraries/**\",\n\t\t\t\t);\nPRODUCT_BUNDLE_IDENTIFIER = ");`
> 
> As a suggestion, why don't you guys change "InsertStaticFrameworkIntoTargetBuildPhaseFrameworks" method to properly use PBXProject class and properly change build properties by just appending the changes that need to be added?

Works for me !

**Comment by rgomezp** [comment]
Thanks for the details. We'll take a look at updating this value

**Comment by LabanLars** [comment]
Any progress on this issue? I have tried all the suggestions here, and I'm still stuck at "library not found for -liPhone-lib" from Xcode.

**Comment by akasurreal** [comment]
> @ArnoBen
> Put it at the end of the OnPostProcessBuild method, just before writing the project file:
> 
> ```
> // Add OneSignal lib paths
> project.AddBuildProperty(notificationServiceTarget, "LIBRARY_SEARCH_PATHS", "$(PROJECT_DIR)/Libraries/OneSignal/Platforms/iOS");
> 
> File.WriteAllText(projectPath, project.WriteToString());
> ```

This worked for me, thanks!

**Comment by jkasten2** [comment]
@NumaNumaNuma Thanks for reporting and digging in to find a working fix. We have implement this fix in PR #251 and it will be included in the next release
